provisioning/transport/amqp/src/amqp.ts

```diff
@@ -73,9 +75,19 @@ export class Amqp extends EventEmitter implements X509ProvisioningTransport, Tpm
       /*Codes_SRS_NODE_PROVISIONING_AMQP_16_017: [The `queryOperationStatus` method shall call its callback with a `RegistrationResult` object parsed from the body of the response message which `correlation_id` matches the `correlation_id` of the request message sent on the sender link.]*/
       const registrationResult = JSON.parse(msg.body.content);
       if (this._operations[msg.correlation_id]) {
+        debug('Got the registration/operationStatus message we were looking for.');
         const requestCallback = this._operations[msg.correlation_id];
         delete this._operations[msg.correlation_id];
-        requestCallback(null, registrationResult, msg, this._config.pollingInterval);
+        let retryAfter: number;
+        /*Codes_SRS_NODE_PROVISIONING_AMQP_06_010: [If the amqp response to a request contains the application property`retry-after`, it will be interpreted as the number of seconds that should elapse before the next attempted operation.  Otherwise default.] */
+        if (msg.application_properties && msg.application_properties[MessagePropertyNames.retryAfter]) {
+          retryAfter = Number(msg.application_properties[MessagePropertyNames.retryAfter]) * 1000;
```

Comment:
>retryAfter [](start = 10, length = 10)

nit: assuming that the *1000 is to turn seconds into milliseconds, it'd be best if the name reflected that: eg. retryAfterInMilliseconds

Reply:
Done.

---
In reply to: [282196964](https://github.com/Azure/azure-iot-sdk-node/pull/555#discussion_r282196964) [](ancestors = 282196964)